src/wallet/test/group_outputs_tests.cpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/setup_common.h>
+
+#include <wallet/coinselection.h>
+#include <wallet/spend.h>
+#include <wallet/wallet.h>
+
+#include <boost/test/unit_test.hpp>
+
+namespace wallet {
+BOOST_FIXTURE_TEST_SUITE(group_outputs_tests, TestingSetup)
+
+static int nextLockTime = 0;
+
+static std::shared_ptr<CWallet> NewWallet(const node::NodeContext& m_node, const ArgsManager& m_args)
+{
+    std::unique_ptr<CWallet> wallet = std::make_unique<CWallet>(m_node.chain.get(), "", m_args, CreateMockWalletDatabase());
+    wallet->LoadWallet();
+    LOCK(wallet->cs_wallet);
+    wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+    wallet->SetupDescriptorScriptPubKeyMans();
+    return wallet;
+}
+
+static void addCoin(CoinsResult& coins,
+                     OutputType type,
+                     CWallet& wallet,
+                     const CScript& dest,
+                     const CAmount& nValue,
+                     bool is_from_me,
+                     CFeeRate fee_rate = CFeeRate(0),
+                     int depth = 6)
+{
+    CMutableTransaction tx;
+    tx.nLockTime = nextLockTime++;        // so all transactions get different hashes
+    tx.vout.resize(1);
+    tx.vout[0].nValue = nValue;
+    tx.vout[0].scriptPubKey = dest;
+
+    const uint256& txid = tx.GetHash();
+    LOCK(wallet.cs_wallet);
+    auto ret = wallet.mapWallet.emplace(std::piecewise_construct, std::forward_as_tuple(txid), std::forward_as_tuple(MakeTransactionRef(std::move(tx)), TxStateInactive{}));
+    assert(ret.second);
+    CWalletTx& wtx = (*ret.first).second;
+    const auto& txout = wtx.tx->vout.at(0);
+    coins.Add(type, {COutPoint(wtx.GetHash(), 0),
+                       txout,
+                       depth,
+                       CalculateMaximumSignedInputSize(txout, &wallet, /*coin_control=*/nullptr),
+                       /*spendable=*/ true,
+                       /*solvable=*/ true,
+                       /*safe=*/ true,
+                       wtx.GetTxTime(),
+                       is_from_me,
+                       fee_rate});
+}
+
+ CoinSelectionParams makeSelectionParams(bool avoid_partial_spends)
+{
+    FastRandomContext rand{};
+    return CoinSelectionParams{
+            rand,
```

Comment:
`CoinSelectionParams`'s `rng_fast` member is a reference and you're using a stack object, this will create a dangling reference.

Reply:
oops, good catch 👍🏼